cypress/support/commands.js

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import stringToHash from '../../pkg/webui/lib/string-to-hash'
+
+// Helper function to quickly login to the oauth app programmatically.
+Cypress.Commands.add('loginOAuth', credentials => {
+  const baseUrl = Cypress.config('baseUrl')
+  const oauthRootPath = Cypress.config('oauthRootPath')
+
+  // Obtain csrf token.
+  cy.request({
+    method: 'GET',
+    url: `${baseUrl}${oauthRootPath}/login`,
+  }).then(({ headers }) => {
+    cy.request({
+      method: 'POST',
+      url: `${baseUrl}${oauthRootPath}/api/auth/login`,
+      body: { user_id: credentials.user_id, password: credentials.password },
+      headers: {
+        'X-CSRF-Token': headers['x-csrf-token'],
+      },
+    })
+  })
+})
+
+// Helper function to quickly login to the console programmatically.
+Cypress.Commands.add('loginConsole', credentials => {
```

Comment:
```suggestion
Cypress.Commands.add('loginConsole', credentials => {
```

Let's use a default argument for the credentials. For most test cases it should not be necessary to use distinct credentials.

Reply:
See https://github.com/TheThingsNetwork/lorawan-stack/pull/2939#issuecomment-662472814